File: src/main/java/smu/nuda/global/batch/exception/CsvValidationException.java
package smu.nuda.global.batch.exception;

public class CsvValidationException extends RuntimeException {

    private final int rowNumber;

    public CsvValidationException(int rowNumber, String message) {
        super("CSV " + rowNumber + "번째 줄: " + message);
        this.rowNumber = rowNumber;
    }

    public int getRowNumber() {
        return rowNumber;
    }
}

File: src/main/java/smu/nuda/global/batch/exception/CsvValidationException.kt
package smu.nuda.global.batch.exception

import smu.nuda.global.error.DomainException
import smu.nuda.global.error.ErrorCode

class CsvValidationException @JvmOverloads constructor(
    errorCode: ErrorCode,
    val rowNumber: Int,
    detailMessage: String = errorCode.message
) : DomainException(
    errorCode,
    buildMessage(rowNumber, detailMessage.ifBlank { errorCode.message })
) {
    companion object {
        @JvmStatic
        private fun buildMessage(rowNumber: Int, message: String): String =
            "CSV ${rowNumber}번째 줄: $message"
    }
}
